### DISCUSSION TITLE
Possible to disable pipeline component by default?

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#13136] [Help: Coding & Implementations] [answered]
It's possible to disable a pipeline component on a loaded model using [Language.disable_pipe](https://spacy.io/api/language#disable_pipe) and similar methods. But, is it possible to configure a pipeline component to be disabled by default on load? I would like to include a pipeline component that is disabled by default, and has to be explicitly enabled.

The effect would be something like this, where a model has a pipeline component named `some_pipe` that has to be explicitly enabled.

```python
nlp = spacy.load("some_model")
print(nlp.pipe_names)
```

```
['tok2vec', 'tagger', 'parser', 'attribute_ruler', 'lemmatizer']
```

```python
nlp.enable_pipe("some_pipe")
print(nlp.pipe_names)
```

```
['tok2vec', 'tagger', 'parser', 'attribute_ruler', 'lemmatizer', 'some_pipe']
```

**Comment by adrianeboyd** [comment, marked as answer]
Sure, just disable the component before saving the pipeline, and it should be disabled by default on load.

The trained pipeline `core` packages all include a disabled `senter`, if it helps to see a concrete example of what the config looks like.

**Reply by kinghuang** [reply]
Whoa, I've never noticed the disabled `senter` component in the `core` packages before! 🤯

Thanks for pointing out the core configs. It made me realize there's a `disabled` config that takes in a list of disabled component names.

```toml
pipeline = ["tok2vec","tagger","parser","attribute_ruler","lemmatizer","ner","llm_ner"]
disabled = ["llm_ner"]
```
